tests/compiletest.rs

```diff
@@ -250,6 +269,21 @@ fn main() -> Result<()> {
     Ok(())
 }
 
+fn run_dep_mode(target: String) -> Result<()> {
+    let files = std::env::args().skip_while(|arg| arg != "--").skip(1);
+    for path in files {
+        let mut config = run_test_config(std::iter::empty(), &target, &path, Mode::Yolo, true);
+        config.program.args.remove(0); // remove the `--error-format=json` argument
+        config.program.args.push("--color".into());
+        config.program.args.push("always".into());
+        let output = ui_test::run_file(config, Path::new(&path))?;
+        std::io::stderr().write_all(&output.stderr)?;
+        std::io::stdout().write_all(&output.stdout)?;
```

Comment:
Is there any way we can forward the output live, rather than buffer and re-emit? In particular this will print the wrong thing when the program prints into both stderr and stdout in an interleaved way.

Reply:
There is no support for that yet, and I'm not sure we want it? Fwiw: stderr and stdout aren't really reliably interleaved on hosts either

Reply:
> stderr and stdout aren't really reliably interleaved on hosts either

I have never seen that not work. On a typical Linux terminal they are both sending things to the same output and they are properly interleaved. It is AFAIK very common for programs to rely on this.

Reply:
I'll have to figure out how to do that, so this won't get fixed in this PR

Reply:
That is because terminals use a single pipe for stdout and stderr and thus are able to preserve relative ordering but can't distinguish between stdout and stderr. You could make a pipe and then pass it to `.stdout()` and `.stderr()` on the `Command` to have the same correct interleaving. `.stdout()` and `.stderr()` accept an `impl Into<Stdio>` which `File` and `OwnedFd` implement.

Reply:
This will get addressed with https://github.com/oli-obk/ui_test/pull/68

Reply:
It wasn't possible to pass arguments without it anyway, so I fixed both this and argument passing in one go.